frame/nomination-pools/src/lib.rs

```diff
@@ -1584,6 +1609,119 @@ pub mod pallet {
+				Error::<T>::NoUnlockChunk
+			);
+
+			// payout related stuff: we must claim the payouts, and updated recorded payout data
+			// before updating the bonded pool points, similar to that of `join` transaction.
+			reward_pool.update_records(bonded_pool.id, bonded_pool.points)?;
+			// TODO: optimize this to not touch the free balance of `who ` at all in benchmarks.
+			// Currently, bonding rewards is like a batch. In the same PR, also make this function
+			// take a boolean argument that make it either 100% pure (no storage update), or make it
+			// also emit event and do the transfer. #11671
+			let _ = Self::do_reward_payout(&who, &mut member, &mut bonded_pool, &mut reward_pool)?;
+
+			// deduct unlocking funds from the pool member and unbonding pool
+			let mut unlocking_balance = BalanceOf::<T>::zero();
+			member.unbonding_eras = member
+				.unbonding_eras
+				.try_mutate(|unbonding_eras| {
+					// rebond from the most recent era (most recent era have longer duration to
+					// withdraw) to the last
+					// TODO: use `BtreeMap::last_entry` once it is stable
+					for (era, points) in unbonding_eras.iter_mut().rev() {
+						let unbonding_pool = sub_pools.get_pool(era);
+						let value = unbonding_pool.point_to_balance(*points);
+
+						if unlocking_balance.saturating_add(value) <= amount {
+							unbonding_pool.dissolve(*points);
+							*points = BalanceOf::<T>::zero();
+							unlocking_balance = unlocking_balance.saturating_add(value);
+						} else {
+							let diff = amount.saturating_sub(unlocking_balance);
+							let diff_points = unbonding_pool.balance_to_point(diff);
+
+							unbonding_pool.dissolve(diff_points);
+							*points = points.saturating_sub(diff_points);
+							unlocking_balance = unlocking_balance.saturating_add(diff);
+						}
+						if unlocking_balance >= amount {
+							break
+						}
+					}
+					sub_pools.with_era.retain(|_, pool| !pool.points.is_zero());
+					unbonding_eras.retain(|_, p| !p.is_zero());
+				})
+				.expect("the length of the map only ever decrease here; qed");
+
+			bonded_pool.ok_to_rebond(&member, unlocking_balance)?;
+
+			// We must calculate the points issued *before* rebond the funds, else points:balance
+			// ratio will be wrong.
+			let points_issued = bonded_pool.issue(unlocking_balance);
+			member.points = member.points.saturating_add(points_issued);
+			// NOTE: it maybe important to rebond the underlying nominator's unlocking chunks
+			// proportionally (according to the member's era unbonding chunks) instead of fully
+			// rebond from the nominator's most recent chunk to the last (i.e. calling
+			// `StakingInterface::rebond` directly), which may lead to some funds become
+			// withdrawable shorter than the `BondingDuration` and thus causing the newly rebond
+			// funds have a larger slash proportion in the proportionally slashing mechanism
+			T::StakingInterface::rebond(bonded_pool.bonded_account(), unlocking_balance)?;
```

Comment:
Proportionally rebond each chunk may bring a bit of complexity. Besides partial rebond unlocking chunks of the nominator, we also need to handle cases like the chunks of some eras had been withdrawn in the nominator account (by other members) but the member who call `rebond` did not, thus we can't rebond the corresponding chunk in the nominator (perhaps calling `withdraw_unbonded` first inside `rebond` can help this case). Not sure if it is worth doing so.